src/components/Accordion/Accordion/Accordion.jsx

```diff
@@ -33,9 +33,12 @@ const Accordion = forwardRef(({ children: originalChildren, allowMultiple, defau
   const onChildClick = useCallback(
     itemIndex => {
       if (allowMultiple) {
-        const newExpandedItems = expandedItems;
+        const newExpandedItems = [...expandedItems];
         if (isChildExpanded(itemIndex)) {
-          newExpandedItems.pop(itemIndex);
+          const index = newExpandedItems.indexOf(itemIndex);
+          if (index > -1) {
+            newExpandedItems.splice(index, 1);
```

Comment:
the second bug was that clicking on already selected drawer caused removing the last opened drawer instead of the one we click on.